baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/manager/DecisionManager.java

```diff
@@ -316,8 +316,12 @@ public Decision deleteDecision(String customerId, String decisionNameOrId) {
 
     private DecisionVersion findDecisionVersion(String decisionId, long version) {
         DecisionVersion decisionVersion = decisionVersionDAO.findByDecisionIdAndVersion(decisionId, version);
+        if (decisionVersion == null) {
+            throw decisionVersionDoesNotExist("", decisionId, version);
```

Comment:
I think the problem of passing an empty string here is that it can make the logs confusing. We'll end up with something like `Version '4' of Decision with id or name 'foo' does not exist for customer ''`. Is the `customerId` missing on purpose or is that another bug in our data model and something has gone wrong elsewhere?

I think we should pass in the `customerId` even if it means doing another DB read as it keeps our logs consistent. Reading through the code, this I think is being caused by the `deleteVersion(String id, long version)` not having the `customerId` as a method parameter. In that case, how about something like this for that method:

```
public DecisionVersion deleteVersion(String id, long version) {
   Decision d = decisionDao.findById(id);
   if(d == null) {
       throw new DecisionNotFoundException("Decision with id '" + id + "' does not exist."); 
   }
   return deleteVersion(d.getCustomerId(), id, version);
}
```

That keeps everything consistent:

- If the Decision doesn't exist, then we get a clear message saying so
- If the Decision version doesn't exist, then we are clearly told which customer and which version doesn't exist

WDYT?

Reply:
@robpblake done, let me know if it looks good to you